### PR TITLE
Migrate xbk

### PR DESCRIPTION
**Pull request by kentico-ericd**
### Motivation

Due to the large amount of code cleanup, optimizations, and features added in the XbK Algolia integration, we would like to migrate that code to the Xperience 13 integration. This is mostly a direct migration of the code, while maintaing Xperience 13-specific functionality such as cultures and sites in indexes.

The migration includes the new data splitting feature from XbK (implements https://github.com/Kentico/xperience-algolia/issues/19), but not the crawler integration.

### Checklist

- [X] Code follows coding conventions held in this repo
- [X] Automated tests have been added
- [X] Tests are passing
- [X] Docs have been updated (if applicable)
- [X] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### CODE REVIEW

File: src/AlgoliaQueueWorker.cs [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+using CMS.Base;
+using CMS.Core;
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
+using Kentico.Xperience.Algolia.Models;
+using Kentico.Xperience.Algolia.Services;
 
-namespace Kentico.Xperience.AlgoliaSearch
```
Comment (seangwright) [inline src/AlgoliaQueueWorker.cs:11]: Will this namespace change mean a major version bump for this library?

Comment (seangwright) [reply src/AlgoliaQueueWorker.cs:11]: Ah, I see the update in the csproj. I usually keep that version information in the `Directory.Build.props` 😀.

File: tests/Kentico.Xperience.AlgoliaSearch.Tests.csproj [hunk 2]
```diff
@@ -1,15 +1,16 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net48</TargetFramework>
-    <RootNamespace>Kentico.Xperience.AlgoliaSearch.Test</RootNamespace>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (kentico-jaroslavn) [inline tests/Kentico.Xperience.AlgoliaSearch.Tests.csproj:4]: I think that project should be multi-targeted to net48 and net6.0

**Review by kentico-jaroslavn — CHANGES_REQUESTED** [review]

File: tests/Kentico.Xperience.AlgoliaSearch.Tests.csproj (hunk 2 above)
Comment (kentico-ericd) [reply tests/Kentico.Xperience.AlgoliaSearch.Tests.csproj]: Fixed in https://github.com/Kentico/xperience-algolia/pull/29/commits/3bdae246678108e671e53255db3733ce7ff53469

File: src/Services/Implementations/DefaultAlgoliaObjectGenerator.cs [hunk 3]
```diff
@@ -217,34 +189,40 @@ protected object GetNodeValue(TreeNode node, PropertyInfo property, Type searchM
...
             {
-                if (prop.DeclaringType == typeof(AlgoliaSearchModel))
-                {
-                    continue;
-                }
+                columnsToUpdate.AddRange(indexedColumns);
+            }
+            else if (queueItem.TaskType == AlgoliaTaskType.UPDATE)
+            {
+                columnsToUpdate.AddRange(queueItem.ChangedColumns.Intersect(indexedColumns));
```
[the remark this answers is not in the document]
Comment (kentico-ericd) [reply src/Services/Implementations/DefaultAlgoliaObjectGenerator.cs:218]: Fixed in https://github.com/Kentico/xperience-algolia/pull/29/commits/3bdae246678108e671e53255db3733ce7ff53469

File: src/Models/AlgoliaIndex.cs [hunk 4]
```diff
@@ -33,9 +44,62 @@ public string IndexName
...
         public IEnumerable<string> SiteNames
+        {
+            get;
+        }
+
+
+        /// <summary>
+        /// An arbitrary ID used to identify the Algolia index in the admin UI.
+        /// </summary>
+        internal int Identifier
```
[the remark this answers is not in the document]
Comment (kentico-ericd) [reply src/Models/AlgoliaIndex.cs]: Fixed in https://github.com/Kentico/xperience-algolia/pull/29/commits/3bdae246678108e671e53255db3733ce7ff53469

File: src/IndexStore.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+
+        internal void Clear()
+        {
+            registeredIndexes.Clear();
+        }
+
+
+        internal AlgoliaIndex Get(int id)
```
[the remark this answers is not in the document]
Comment (kentico-ericd) [reply src/IndexStore.cs]: Fixed in https://github.com/Kentico/xperience-algolia/pull/29/commits/3bdae246678108e671e53255db3733ce7ff53469

File: src/AlgoliaQueueWorker.cs [hunk 6]
```diff
@@ -29,54 +29,53 @@ public class AlgoliaQueueWorker : ThreadQueueWorker<AlgoliaQueueItem, AlgoliaQue
...
+            algoliaClient = Service.Resolve<IAlgoliaClient>();
         }
 
 
         /// <summary>
         /// Adds an <see cref="AlgoliaQueueItem"/> to the worker queue to be processed.
         /// </summary>
         /// <param name="queueItem">The item to be added to the queue.</param>
+        /// <exception cref="InvalidOperationException" />
         public static void EnqueueAlgoliaQueueItem(AlgoliaQueueItem queueItem)
```
[the remark this answers is not in the document]
Comment (kentico-ericd) [reply src/AlgoliaQueueWorker.cs]: Fixed in https://github.com/Kentico/xperience-algolia/pull/29/commits/3bdae246678108e671e53255db3733ce7ff53469

**Review by kentico-jaroslavn — APPROVED** [review]

**Review by kentico-jaroslavn — APPROVED** [review]

**Pull request merged**